File: sea-shell-service-spring-web-flux/sea-shell-rest-service/src/test/java/org/jesperancinha/shell/SeaShellReactiveOneServiceImplTest.java
package org.jesperancinha.shell;

import com.github.tomakehurst.wiremock.WireMockServer;
import org.jesperancinha.shell.webflux.service.SeaShellReactiveOneServiceImpl;
import org.junit.jupiter.api.AfterAll;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import reactor.blockhound.BlockHound;

import java.io.IOException;
import java.time.Duration;

import static org.junit.jupiter.api.Assertions.assertAll;
import static reactor.core.publisher.Mono.delay;

@SpringBootTest
public class SeaShellReactiveOneServiceImplTest {

    private static WireMockServer wireMockServer;

    @Autowired
    private SeaShellReactiveOneServiceImpl seaShellReactiveOneService;

    @BeforeAll
    public static void setUpAll() throws IOException {
        BlockHound.install();
        wireMockServer = SeaShellWiremockSoapLauncher.createWireMockServer();
    }

    @Test
    public void findAllIds_onCall_thenNonBlocking() {
        assertAll(() -> delay(Duration.ofSeconds(1))
                .doOnNext(it -> seaShellReactiveOneService.getAllIds())
                .block());
    }

    @Test
    public void findSeaShellById_onCall_thenNonBlocking() {
        assertAll(() -> delay(Duration.ofSeconds(1))
                .doOnNext(it -> seaShellReactiveOneService.getSeaShellById(1L))
                .block());
    }

    @Test
    public void findPersonById_onCall_thenNonBlocking() {
        assertAll(() -> delay(Duration.ofSeconds(1))
                .doOnNext(it -> seaShellReactiveOneService.getPersonById(1L))
                .block());
    }

    @Test
    public void findCostumeById_onCall_thenNonBlocking() {
        assertAll(() -> delay(Duration.ofSeconds(1))
                .doOnNext(it -> seaShellReactiveOneService.getCostumeById(1L))
                .block());
    }

    @Test
    public void findAccountById_onCall_thenNonBlocking() {
        assertAll(() -> delay(Duration.ofSeconds(1))
                .doOnNext(it -> seaShellReactiveOneService.getAccountById("ACCOUNTID"))
                .block());
    }

    @Test
    public void findTopById_onCall_thenNonBlocking() {
        assertAll(() -> delay(Duration.ofSeconds(1))
                .doOnNext(it -> seaShellReactiveOneService.getTopById(1L))
                .block());
    }

    @Test
    public void findLowerById_onCall_thenNonBlocking() {
        assertAll(() -> delay(Duration.ofSeconds(1))
                .doOnNext(it -> seaShellReactiveOneService.getLowerById(1L))
                .block());
    }

    @AfterAll
    public static void afterAll() {
        wireMockServer.stop();
    }
}

File: sea-shell-service-spring-web-flux/src/test/kotlin/org/jesperancinha/shell/SeaShellReactiveOneServiceImplTest.kt
package org.jesperancinha.shell

import com.github.tomakehurst.wiremock.WireMockServer
import org.jesperancinha.shell.webflux.service.SeaShellReactiveOneServiceImpl
import org.junit.jupiter.api.AfterAll
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeAll
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.function.Executable
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import reactor.blockhound.BlockHound
import reactor.core.publisher.Mono
import java.io.IOException
import java.time.Duration

@SpringBootTest
class SeaShellReactiveOneServiceImplTest {
    @Autowired
    private val seaShellReactiveOneService: SeaShellReactiveOneServiceImpl? = null
    @Test
    fun findAllIds_onCall_thenNonBlocking() {
        Assertions.assertAll(Executable {
            Mono.delay(Duration.ofSeconds(1))
                .doOnNext { it: Long? -> seaShellReactiveOneService!!.allIds }
                .block()
        })
    }

    @Test
    fun findSeaShellById_onCall_thenNonBlocking() {
        Assertions.assertAll(Executable {
            Mono.delay(Duration.ofSeconds(1))
                .doOnNext { it: Long? -> seaShellReactiveOneService!!.getSeaShellById(1L) }
                .block()
        })
    }

    @Test
    fun findPersonById_onCall_thenNonBlocking() {
        Assertions.assertAll(Executable {
            Mono.delay(Duration.ofSeconds(1))
                .doOnNext { it: Long? -> seaShellReactiveOneService!!.getPersonById(1L) }
                .block()
        })
    }

    @Test
    fun findCostumeById_onCall_thenNonBlocking() {
        Assertions.assertAll(Executable {
            Mono.delay(Duration.ofSeconds(1))
                .doOnNext { it: Long? -> seaShellReactiveOneService!!.getCostumeById(1L) }
                .block()
        })
    }

    @Test
    fun findAccountById_onCall_thenNonBlocking() {
        Assertions.assertAll(Executable {
            Mono.delay(Duration.ofSeconds(1))
                .doOnNext { it: Long? -> seaShellReactiveOneService!!.getAccountById("ACCOUNTID") }
                .block()
        })
    }

    @Test
    fun findTopById_onCall_thenNonBlocking() {
        Assertions.assertAll(Executable {
            Mono.delay(Duration.ofSeconds(1))
                .doOnNext { it: Long? -> seaShellReactiveOneService!!.getTopById(1L) }
                .block()
        })
    }

    @Test
    fun findLowerById_onCall_thenNonBlocking() {
        Assertions.assertAll(Executable {
            Mono.delay(Duration.ofSeconds(1))
                .doOnNext { it: Long? -> seaShellReactiveOneService!!.getLowerById(1L) }
                .block()
        })
    }

    companion object {
        private var wireMockServer: WireMockServer? = null
        @BeforeAll
        @Throws(IOException::class)
        fun setUpAll() {
            BlockHound.install()
            wireMockServer = SeaShellWiremockSoapLauncher.createWireMockServer()
        }

        @AfterAll
        fun afterAll() {
            wireMockServer!!.stop()
        }
    }
}